torchtext/datasets/language_modeling.py

```diff
@@ -1,217 +1,213 @@
-from .. import data
+import torch
+import logging
 import io
+import os
+from torchtext.utils import download_from_url, extract_archive, unicode_csv_reader
+from torchtext.vocab import build_vocab_from_iterator
+from torchtext.data.utils import get_tokenizer
+from torchtext.vocab import Vocab
+from tqdm import tqdm
+
+URLS = {
+    'WikiText2':
+        'https://s3.amazonaws.com/research.metamind.io/wikitext/wikitext-2-v1.zip',
+    'WikiText103':
+        'https://s3.amazonaws.com/research.metamind.io/wikitext/wikitext-103-v1.zip',
+    'PennTreebank':
+        ['https://raw.githubusercontent.com/wojzaremba/lstm/master/data/ptb.train.txt',
+         'https://raw.githubusercontent.com/wojzaremba/lstm/master/data/ptb.test.txt',
+         'https://raw.githubusercontent.com/wojzaremba/lstm/master/data/ptb.valid.txt']
+}
+
+
+def _read_text_iterator(data_path, tokenizer):
```

Comment:
These two functions look similar to what's in the classification datasets - is there a way to share this code via some simple abstractions?